src/main/scala/com/databricks/spark/xml/parsers/StaxXmlGenerator.scala

```diff
@@ -46,9 +46,26 @@ private[xml] object StaxXmlGenerator {
         // If this is meant to be value but in no child, write only a value
         writeElement(dt, v)
       case (_, _, _) =>
-        writer.writeStartElement(name)
-        writeElement(dt, v)
-        writer.writeEndElement()
+        var hasChild = true
+        if (options.selfCloseTag == "true" && dt.isInstanceOf[StructType]){
```

Comment:
reading self-closing tag is fixed. If there's an issue, it should be fixed in XML format rather than here for performance.

Reply:
Oops, yea, I rushed to read.

Reply:
but is it really a proper way to read self-closing tag? Using the regular expression looks not quite a great solution and I think it's okay to write out non-self-closing tag since both are legitimate XML element format.

Reply:
because our customer wants us use self close tag, that's why I add this enhancement.  I know both are legitimate

Reply:
they think it is redundant

Reply:
It's okay if we can read and write back. Let's don't add it for now.

Reply:
Yes, I don't think we need this with this regular expression way. If there's simpler and robust way, that's okay but current way looks too hacky and writing out self-closing tags looks not super important given that the reason is that it just looks redundant.